app/page.tsx

```diff
@@ -2,29 +2,201 @@
 import { useAuth } from "./AuthContext";
 import TopBar from "@/components/TopBar";
 import BottomNavigationBar from "@/components/BottomNavigationBar";
+import { getSessionOccurrences, getSessions } from "./SessionList";
+import { useState, useEffect } from "react";
+import { LiveClasses } from "./types";
+import Link from "next/link";
+import PrimaryButton from "@/components/Button";
+import Loading from "./loading";
 
 export default function Home() {
   const { loggedIn, userId } = useAuth();
+  const [liveClasses, setLiveClasses] = useState<LiveClasses[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [quizzes, setQuizzes] = useState<LiveClasses[]>([]);
+  const baseUrl = process.env.NEXT_PUBLIC_AF_QUIZ_URL;
+  const apiKey = process.env.NEXT_PUBLIC_AF_QUIZ_API_KEY;
+  const userID = process.env.NEXT_PUBLIC_AF_QUIZ_USER_ID;
+
+  const fetchSessionOccurrencesAndDetails = async () => {
+    try {
+      const sessionOccurrenceData = await getSessionOccurrences();
+      const currentDate = new Date();
+      currentDate.setHours(0, 0, 0, 0);
+      const endOfWeek = new Date();
+      endOfWeek.setDate(endOfWeek.getDate() + (6 - endOfWeek.getDay()));
+
+      const todaySessions = [];
+
+      for (const sessionOccurrence of sessionOccurrenceData) {
+        const sessionStartTime = new Date(sessionOccurrence.start_time);
+
+        if (sessionStartTime > currentDate && sessionStartTime <= endOfWeek) {
+          try {
+            const sessionDetail = await getSessions(sessionOccurrence.session_fk);
+            if (isSameDay(sessionStartTime, currentDate)) {
+              todaySessions.push({
```

Comment:
What if it's not same day? Why fetch till end of week if we are displaying only today's sessions

Reply:
Yeah correct. Previously I was thinking to show sessions happening in the current week. But we will be showing sessions happening in a day. Will make the changes